domain/optatask/pom.xml

```diff
@@ -8,27 +8,51 @@
   <name>optatask</name>
 
   <properties>
-    <version.org.optaplanner>7.12.0.Final</version.org.optaplanner>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
     <version.org.kie>7.12.0.Final</version.org.kie>
   </properties>
 
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.kie</groupId>
+        <artifactId>kie-parent</artifactId>
+        <version>${version.org.kie}</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+      <dependency>
+        <groupId>org.optaplanner</groupId>
+        <artifactId>optaplanner-bom</artifactId>
+        <version>${version.org.kie}</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+
   <dependencies>
```

Comment:
`mvn test` is throwing this error:
```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.084 s <<< FAILURE! - in org.optatask.domain.TaskAssigningRealtimeTest
[ERROR] addAndRemoveTaskProblemFactChange(org.optatask.domain.TaskAssigningRealtimeTest)  Time elapsed: 0.002 s  <<< ERROR!
java.lang.NoClassDefFoundError: ch/qos/logback/core/util/StatusListenerConfigHelper
	at org.optatask.domain.TaskAssigningRealtimeTest.<init>(TaskAssigningRealtimeTest.java:42)
Caused by: java.lang.ClassNotFoundException: ch.qos.logback.core.util.StatusListenerConfigHelper
	at org.optatask.domain.TaskAssigningRealtimeTest.<init>(TaskAssigningRealtimeTest.java:42)

```
Adding the following dependency solved the issue:
```xml
    <dependency>
      <groupId>ch.qos.logback</groupId>
      <artifactId>logback-core</artifactId>
      <version>1.2.3</version>
      <scope>test</scope>
    </dependency>
```
Why do you think it's thrown now?

Reply:
It's because I'm importing `kie-parent` (BOM) where `logback-classic` dependency is managed and its version is set to 1.1.3 which is unsurprisingly not compatible with `logback-core:1.2.3`.

I didn't realize that both `junit` and `logback-core` became managed. The correct approach is not to override the managed version.